src/AppInstallerCLICore/Workflows/InstallFlow.cpp

```diff
@@ -248,4 +261,128 @@ namespace AppInstaller::CLI::Workflow
+        {
+            context.Reporter.Error() << Resource::String::MSStoreInstallStoreClientBlocked << std::endl;
+            AICLI_LOG(CLI, Error, << "Store client is blocked by policy. MSStore install failed.");
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_MSSTORE_BLOCKED_BY_POLICY);
+        }
+
+        if (!installManager.GetIsAppAllowedToInstallAsync(productId).get())
+        {
+            context.Reporter.Error() << Resource::String::MSStoreInstallAppBlocked << std::endl;
+            AICLI_LOG(CLI, Error, << "App is blocked by policy. MSStore install failed. ProductId: " << Utility::ConvertToUTF8(productId));
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_MSSTORE_APP_BLOCKED_BY_POLICY);
+        }
+
+        // Verifying/Acquiring product ownership
+        context.Reporter.Info() << Resource::String::MSStoreInstallTryGetEntitlement << std::endl;
+        GetEntitlementResult enr = installManager.GetFreeUserEntitlementAsync(productId, winrt::hstring(), winrt::hstring()).get();
+
+        if (enr.Status() == GetEntitlementStatus::Succeeded)
+        {
+            context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementSuccess << std::endl;
+            AICLI_LOG(CLI, Error, << "Get entitlement succeeded.");
+        }
+        else
+        {
+            if (enr.Status() == GetEntitlementStatus::NoStoreAccount)
+            {
+                context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementNoStoreAccount << std::endl;
+                AICLI_LOG(CLI, Error, << "Get entitlement failed. No Store account.");
+            }
+            else if (enr.Status() == GetEntitlementStatus::NetworkError)
+            {
+                context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementNetworkError << std::endl;
+                AICLI_LOG(CLI, Error, << "Get entitlement failed. Network error.");
+            }
+            else if (enr.Status() == GetEntitlementStatus::ServerError)
+            {
+                context.Reporter.Info() << Resource::String::MSStoreInstallGetEntitlementServerError << std::endl;
+                AICLI_LOG(CLI, Error, << "Get entitlement succeeded. Server error. ProductId: " << Utility::ConvertToUTF8(productId));
+            }
+
+            AICLI_TERMINATE_CONTEXT(APPINSTALLER_CLI_ERROR_MSSTORE_INSTALL_FAILED);
+        }
+
+        context.Reporter.Info() << Resource::String::InstallFlowStartingPackageInstall << std::endl;
+
+        IVectorView<AppInstallItem> installItems = installManager.StartProductInstallAsync(
+            productId,              // ProductId
+            winrt::hstring(),       // CatalogId
+            winrt::hstring(),       // FlightId
+            L"WinGetCli",           // ClientId
+            false,                  // repair
+            false,
+            winrt::hstring(),
+            nullptr).get();
+
+        auto installItem = installItems.GetAt(0);

```

Comment:
>auto installItem = installItems.GetAt(0); [](start = 8, length = 41)

Do we not need to handle all of the items?  A comment is probably required here.

Reply:
From my reading of the docs, if the target is a bundle this will have every package.  I think we do need to watch all of them.

---
In reply to: [454004580](https://github.com/Microsoft/pkg/pull/493#discussion_r454004580) [](ancestors = 454004580)